### PR TITLE
Don't ignore the linter

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: conda_forge_tick_action/automerge.py [hunk 1]
```diff
@@ -13,8 +13,8 @@
 # github actions use the check_suite API
 IGNORED_CHECKS = ['github-actions']
 
-# always ignore the linter since it is not reliable
```
Comment (isuruf) [inline conda_forge_tick_action/automerge.py:16]: If the linter is not reliable, we should fix it. I don't see anything about it at https://github.com/conda-forge/conda-smithy/issues?q=is%3Aissue+is%3Aopen+lint

Comment (beckermr) [reply conda_forge_tick_action/automerge.py:16]: I agree in principle, but in practice, the linter being flaky is not a good enough reason IMO for us to not allow the bot or a maintainer to merge a PR via automerge.

I'd support adding a linter required key to the conda-forge.yml that would let people turn on "strict" status checks, but I do think the default should be to ignore it.

Comment (isuruf) [reply conda_forge_tick_action/automerge.py:16]: Where are these issues about linter being flaky?
I disagree about opt in for linter. It should be opt out as it is the case when humans merge.

Comment (beckermr) [reply conda_forge_tick_action/automerge.py:16]: To address your other point, right now we have no issues, but historically this has not been the case. Further, we don't ship conda-smithy releases fast enough to not interrupt a migration with automerge should a linter error arise. Finally, there could be legitimate reasons a maintainer might ignore the linter.

Comment (beckermr) [reply conda_forge_tick_action/automerge.py:16]: I just ignore the problems when I find them. It was missing tests for a while, for example.

Comment (isuruf) [reply conda_forge_tick_action/automerge.py:16]: > To address your other point, right now we have no issues, but historically this has not been the case.

True, but that's extra motivation to fix them.

> Further, we don't ship conda-smithy releases fast enough to not interrupt a migration with automerge should a linter error arise.

We can release conda-smithy fast. conda-smithy being slow to release is not an issue we have seen either.

>  Finally, there could be legitimate reasons a maintainer might ignore the linter.

I've added a opt out.

**Review by beckermr — CHANGES_REQUESTED** [review]
Thanks for the edits @isuruf. I still don't agree with this change. I suggest we ice this PR until we both have had a chance to think more on these issues.

**Comment by isuruf** [comment]
I still don't know what these issues are. Please open them at conda-smithy repo and I'll fix them.

**Comment by beckermr** [comment]
Sure. I'm happy to dig around. Some of them may have already been fixed.

**Comment by isuruf** [comment]
@conda-forge/core, this is a serious issue that we need a decision on.

**Review by beckermr — APPROVED** [review]

**Pull request merged**